runelite-client/src/main/java/net/runelite/client/plugins/npcunaggroarea/NpcAggroAreaPlugin.java

```diff
@@ -0,0 +1,579 @@
+		String npcName = composition.getName().toLowerCase();
+		if (npcLvl > 0 && playerLvl > npcLvl * 2)
+		{
+			return false;
+		}
+
+		for (String pattern : npcNamePatterns)
+		{
+			if (WildcardMatcher.matches(pattern, npcName))
+			{
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	private void recheckActive()
+	{
+		active = config.alwaysActive();
+
+		if (!active)
+		{
+			for (NPC npc : client.getNpcs())
+			{
+				if (isNpcMatch(npc))
+				{
+					active = true;
+					break;
+				}
+			}
+		}
+
+		reevaluateActive();
+	}
+
+	@Subscribe
+	public void onNpcSpawned(NpcSpawned event)
+	{
+		if (config.alwaysActive())
+		{
+			return;
+		}
+
+		if (!active && isNpcMatch(event.getNpc()))
+		{
+			active = true;
+			reevaluateActive();
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		WorldPoint newLocation = client.getLocalPlayer().getWorldLocation();
+		if (lastPlayerLocation != null)
+		{
+			if (safeCenters[1] == null && newLocation.distanceTo2D(lastPlayerLocation) > SAFE_AREA_RADIUS * 4)
+			{
+				safeCenters[0] = null;
+				safeCenters[1] = newLocation;
+				resetTimer();
+				calculateLinesToDisplay();
+
+				// We don't know where the previous area was, so if the player e.g.
+				// entered a dungeon and then goes back out, he/she may enter the previous
+				// area which is unknown and would make the plugin inaccurate
+				previousUnknownCenter = lastPlayerLocation;
+			}
+		}
+
+		if (safeCenters[0] == null && previousUnknownCenter != null &&
+			previousUnknownCenter.distanceTo2D(newLocation) <= UNKNOWN_AREA_RADIUS)
+		{
+			// Player went back to their previous unknown area before the 2nd
+			// center point was found, which means we don't know where it is again.
+			safeCenters[1] = null;
+			endTime = null;
+			updateTimer();
+			calculateLinesToDisplay();
+		}
+
+		if (safeCenters[1] != null)
+		{
+			if (Arrays.stream(safeCenters).noneMatch(
+				x -> x != null && x.distanceTo2D(newLocation) <= SAFE_AREA_RADIUS))
+			{
+				safeCenters[0] = safeCenters[1];
+				safeCenters[1] = newLocation;
+				resetTimer();
+				calculateLinesToDisplay();
+				previousUnknownCenter = null;
+			}
+		}
+
+		lastPlayerLocation = newLocation;
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		String key = event.getKey();
+		switch (key)
+		{
+			case "npcUnaggroAlwaysActive":
+				recheckActive();
+				break;
+			case "npcUnaggroShowTimer":
+				updateTimer();
+				break;
+			case "npcUnaggroCollisionDetection":
+			case "npcUnaggroShowAreaLines":
+				calculateLinesToDisplay();
+				break;
+			case "npcUnaggroNames":
+				npcNamePatterns = NAME_SPLITTER.splitToList(config.npcNamePatterns());
+				recheckActive();
+				break;
+		}
+	}
+
+	private void loadConfig()
+	{
+		String configGroup = NpcAggroAreaConfig.CONFIG_GROUP + "." + username;
+
+		safeCenters[0] = configManager.getConfiguration(configGroup, NpcAggroAreaConfig.CONFIG_CENTER1, WorldPoint.class);
```

Comment:
All of this just for account specific config? Probably not worth it.

Reply:
It's mostly there so the area and timer can be determined upon login. If we remove the config, the area will be unknown on login and the user would have to teleport/enter a dungeon before the plugin figures out where to outline the area.

Reply:
The config is fine - but it would be a lot simpler with using the normal config interface... the only reason to use config manager in this way is for login-specific config, which is sort of a hack.

Reply:
Well if the user logs onto another account, the config for the other account would need to be removed in that case, causing the same issue. As far as I know the only difference is adding the username to the config anyway, so it's not really more complicated?

Reply:
We already do this at other places and trying to migrate it from global config to account-specific in future would be more annoying than having it account specific in first place, even with "hacky" method imo.